cg/services/orders/store_order_services/store_fastq_order_service.py

```diff
@@ -33,84 +33,109 @@ def store_order(self, order: FastqOrder) -> dict:
+            ordered=datetime.now(),
+            original_ticket=str(ticket_id),
+            priority=sample.priority,
+            tumour=sample.tumour,
+            capture_kit=sample.capture_kit,
+            subject_id=sample.subject_id,
             customer=customer,
-            ticket_id=ticket_id,
+            application_version=application_version,
+            order=order_name,
         )
-        with self.status_db.session.no_autoflush:
-            for sample in order.samples:
-                new_sample = self.status_db.add_sample(
-                    name=sample.name,
-                    sex=sample.sex or "unknown",
-                    comment=sample.comment,
-                    internal_id=sample._generated_lims_id,
-                    order=order.name,
-                    ordered=datetime.now(),
-                    original_ticket=str(ticket_id),
-                    priority=sample.priority,
-                    tumour=sample.tumour,
-                    capture_kit=sample.capture_kit,
-                    subject_id=sample.subject_id,
-                )
-                new_sample.customer = customer
-                application_tag: str = sample.application
-                application_version: ApplicationVersion = (
-                    self.status_db.get_current_application_version_by_tag(application_tag)
-                )
-                new_sample.application_version = application_version
-                new_samples.append(new_sample)
-                if not case:
-                    case = self.status_db.add_case(
-                        data_analysis=Workflow.RAW_DATA,
-                        data_delivery=DataDelivery.FASTQ,
-                        name=str(ticket_id),
-                        priority=sample.priority,
-                        ticket=str(ticket_id),
-                    )
-                if (
-                    not new_sample.is_tumour
-                    and new_sample.prep_category == SeqLibraryPrepCategory.WHOLE_GENOME_SEQUENCING
-                ):
-                    self.create_maf_case(sample_obj=new_sample, order=status_db_order)
-                case.customer = customer
-                new_relationship = self.status_db.relate_sample(
-                    case=case, sample=new_sample, status=StatusEnum.unknown
-                )
-                self.status_db.session.add_all([case, new_relationship])
-        status_db_order.cases.append(case)
-        self.status_db.session.add(status_db_order)
-        self.status_db.session.add_all(new_samples)
-        self.status_db.session.commit()
-        return new_samples
+
+    def _create_maf_case(self, db_sample: Sample, db_order: Order) -> None:
+        """
+        Add a MAF case to the current Status database transaction only if the given sample is
+        non-tumour and  WGS. Return None otherwise without adding anything to the transaction.
+        This function does not commit to the database.
+        """
```

Comment:
The method seems to be both creating and linking the case so maybe we should mention that in the docstring. Also - would we want to call it "ensure_maf_case" or something given that it might not add anything?
```suggestion
    def _create_maf_case(self, db_sample: Sample, db_order: Order) -> None:
        """
        Add a MAF case to the current Status database transaction and link it to the given sample, but only if the given sample is
        non-tumour and  WGS. 
        This function does not commit to the database.
        """
```